build-infrastructure/al-maintenance-codebuild-stack.yml

```diff
@@ -0,0 +1,59 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: A template that brings up a code build project that makes release configs for AL versions in maintenance mode
+
+Parameters:
+    GithubFullRepoName:
+      Type: String
+      Description: Full name of the agent repository to use (e.g. https://github.com/aws/amazon-ecs-agent.git)
+      Default: https://github.com/aws/amazon-ecs-agent.git
+    GithubBranchName:
+      Type: String
+      Description: Name of the base branch to use to build, PRs to which will trigger builds (e.g. refs/heads/AmazonLinuxLegacy)
+    GithubSourceBranchName:
+      Type: String
+      Description: Name of the source branch used for the PRs (e.g. al1-patch)
+    ReleaseArtifactsBucketS3Uri:
+      Type: String
+      Description: The URI of the bucket where things land at the end, this is assumed to already exist (e.g. s3://artifacts)
+    ServiceRoleArn:
+      Type: String
+      Description: The ARN of an existing CodeBuild service role
+
+Resources:
+  ModifyAndCopyJSON:
+    Type: 'AWS::CodeBuild::Project'
+    Properties:
+      Artifacts:
+        Type: NO_ARTIFACTS
+      BadgeEnabled: false
+      ConcurrentBuildLimit: 10
+      Description: A CodeBuild project to modify and update the release configs for AL versions in maintenance mode. Builds are triggered by PR merges.
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        ImagePullCredentialsType: CODEBUILD
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
```

Comment:
nit: Should we use [aws/codebuild/amazonlinux2-x86_64-standard:4.0](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html)? This might be the latest AL2 image they are offering?

Reply:
If that's the case then we should update all of our CodePipeline/CodeBuild to use the latest AL2 image. I kept it as version 3.0 to keep it the same as our other projects.

Reply:
Perhaps this could be a follow up.